api/base/serializers.py

```diff
@@ -143,92 +171,142 @@ def get_attribute(self, obj):
+        return meta
 
-            meta[key] = _rapply(self.meta[key], _url_val, obj=value, serializer=self.parent)
+    def lookup_attribute(self, obj, lookup_field):
+        """
+        Returns attribute from target object unless attribute surrounded in angular brackets where it returns the lookup field.
 
-        return {'links': {self.link_type: {'href': url, 'meta': meta}}}
+        Also handles the lookup of nested attributes.
+        """
+        bracket_check = _tpl(lookup_field)
+        if bracket_check:
+            source_attrs = bracket_check.split('.')
+            return get_nested_attributes(obj, source_attrs)
+        return lookup_field
 
+    def kwargs_lookup(self, obj, kwargs_dict):
+        """
+        For returning kwargs dictionary of format {"lookup_url_kwarg": lookup_value}
+        """
+        kwargs_retrieval = {}
+        for lookup_url_kwarg, lookup_field in kwargs_dict.items():
+            try:
+                lookup_value = self.lookup_attribute(obj, lookup_field)
+            except AttributeError as exc:
+                raise AssertionError(exc)
+            if lookup_value is None:
+                return None
+            kwargs_retrieval[lookup_url_kwarg] = lookup_value
+        return kwargs_retrieval
 
-class JSONAPIHyperlinkedRelatedField(ser.HyperlinkedRelatedField):
-    """
-    HyperlinkedRelated field that returns a nested dict with url,
-    optional meta information, and link_type.
+    # Overrides HyperlinkedIdentityField
+    def get_url(self, obj, view_name, request, format):
+        urls = {}
+        for view_name, view in self.views.items():
+            if view is None:
+                urls[view_name] = {}
+            else:
+                kwargs = self.kwargs_lookup(obj, self.view_kwargs[view_name])
+                if kwargs is None:
+                    urls[view_name] = {}
+                else:
+                    urls[view_name] = self.reverse(view, kwargs=kwargs, request=request, format=format)
 
-    Example:
+        if not urls['self'] and not urls['related']:
+            urls = {}
 
-        branched_from = JSONAPIHyperlinkedRelatedField(view_name='nodes:node-detail', lookup_field='pk',
-                                                    lookup_url_kwarg='node_id', read_only=True, link_type='related')
+        return urls
 
-    """
-    json_api_link = True  # serializes to a links object
+    # Overrides HyperlinkedIdentityField
+    def to_representation(self, value):
 
-    def __init__(self, view_name=None, **kwargs):
-        self.meta = kwargs.pop('meta', {})
-        self.link_type = kwargs.pop('link_type', 'url')
-        super(JSONAPIHyperlinkedRelatedField, self).__init__(view_name=view_name, **kwargs)
+        urls = super(RelationshipField, self).to_representation(value)
 
-    def to_representation(self, value):
-        """
-        Returns nested dictionary in format {'links': {'self.link_type': ... }
+        if urls == {}:
```

Comment:
It seems more straightforward to allow `get_url` to return `None` rather than having it return `{}` and then returning `None` in `to_representation`.

I can change this when I merge.